01_Data/src/layers/sequelize/repository/TransactionEvent.ts

```diff
@@ -143,11 +155,74 @@ export class SequelizeTransactionEventRepository extends SequelizeRepository<Tra
     });
   }
 
+  private async calculateAndUpdateTotalKwh(transaction: Transaction): Promise<number> {
+    const totalKwh = this.getTotalKwh(transaction.meterValues ?? []);
+
+    await transaction.update(
+      {totalKwh}
+    );
+
+    return totalKwh;
+  }
+
+  private getTotalKwh(meterValues: MeterValueType[]): number {
+    const contexts: ReadingContextEnumType[] = [
+      ReadingContextEnumType.Transaction_Begin,
+      ReadingContextEnumType.Sample_Periodic,
+      ReadingContextEnumType.Transaction_End,
+    ];
+
+    let valuesMap = new Map();
+
+    meterValues
+      .filter(
+        (meterValue) =>
+          meterValue.sampledValue[0].context &&
```

Comment:
when absent, context is Sample.Periodic

From spec, definition of context field: 
> Type of detail value: start, end or sample.
Default = "Sample.Periodic"

so you should actually make it:
> !meterValue.sampledValue[0].context ||